src/rsvc/srv.c

```diff
@@ -1044,7 +1044,7 @@ ds_rsvc_add_replicas_s(struct ds_rsvc *svc, d_rank_list_t *ranks, size_t size)
 				true /* create */, false /* bootstrap */, size);
 
 	/* TODO: Attempt to only add replicas that were successfully started */
-	if (rc != 0)
+	if (rc != 0 && rc != -DER_ALREADY)
```

Comment:
Looks good.

Not for this patch but I am wondering if inside ds_rsvc_dist_start() when -DER_ALREADY is seen from the RSVC_START, if it can skip the call to ds_rsvc_dist_stop() before returning -DER_ALREADY to the caller (this code). Or if that must still be done even though it is likely to also result in a -DER_ALREADY (and be converted to rc=0 in ds_rsvc_stop_handler()).

Reply:
Ah, you're right, @kccain. The current patch doesn't work correctly when I tried it; I'll help fix it in a moment...

Reply:
can this change actually help? in ds_rsvc_dist_start(), we did this as well:
```
        out = crt_reply_get(rpc);
        rc = out->sao_rc;
        if (rc != 0) {
                D_ERROR(DF_UUID": failed to start%s %d replicas: "DF_RC"\n",
                        DP_UUID(dbid), create ? "/create" : "", rc,
                        DP_RC(out->sao_rc_errval));
                ds_rsvc_dist_stop(class, id, ranks, NULL, create);
                rc = out->sao_rc_errval;
        }
```
so we still calls ds_rsvc_dist_stop() for DER_ALREADY,  but now we will ignore it and continue rdb_add_replicas(), which is incorrect to me

Reply:
It did not work when I tried it; I'm testing a fix...